lab_3_ann_retriever/main.py

```diff
@@ -220,6 +309,39 @@ def _calculate_tf_idf(self, document: list[str]) -> Vector | None:
 
         In case of corrupt input arguments, None is returned.
         """
+        if not isinstance(document, list) or not all(isinstance(item, str) for item in document) or not document:
+            return None
+
+        tf = {}
+
+        voc = set(self._vocabulary)
+        doc = set(document)
+
+        for elem in voc.union(doc):
+            tf[elem] = document.count(elem) / len(document)
+
+        return tuple((tf[elem] * self._idf_values[elem] if elem in document else 0.0 for elem in self._vocabulary))
```

Comment:
try to simplify

Reply:
Done. I've added the second option, but I don't like it and don't know why, though it works too